Supports reading and writing HostName on ESP8266

MRT-77 opened the pull request:
```
ARM_DRIVER_WIFI *wifi = &Driver_WiFi0;

// Set
wifi->SetOption(0, ARM_WIFI_HOSTNAME, "Test", 4);

uint32_t len = 32;
char buf[len];

// Get
wifi->GetOption(0, ARM_WIFI_HOSTNAME, buf, &len);
```

MRT-77 committed:
added functions prototype

function AT_Cmd_HostName and AT_Resp_HostName

MRT-77 committed:
functions for support HostName

function AT_Resp_HostName and AT_Cmd_ConnectAP

MRT-77 committed:
add AT command CWHOSTNAME

added to List_PlusResp and CommandCode_t

MRT-77 committed:
case ARM_WIFI_HOSTNAME added

Added to functions ARM_WIFI_GetOption and ARM_WIFI_SetOption.

MRT-77 committed:
fix ARM_WIFI_GetOption -> case ARM_WIFI_HOSTNAME

VladimirUmek commented:
Hi,

this is a very nice contribution but please bear in mind that we must follow CMSIS API specification. CMSIS pull request https://github.com/ARM-software/CMSIS_5/pull/933 must be therefore discussed and resolved first before we will know how to proceed with this one.

Many thanks for your proposal!

MRT-77 committed:
edit function prototype -> AT_Cmd_ConnectAP

MRT-77 committed:
edit function AT_Cmd_ConnectAP

1_ char out[113+1]; because 32 char SSID, 64 char Password, 17 char BSSID.
2_ print bssid in "xx:xx:xx:xx:xx:xx" format.

MRT-77 committed:
use bssid in WIFI_Activate and WIFI_GetNetInfo

MRT-77 commented:
The latest change is to use **BSSID**.
If you have multiple WiFi with a similar name, you can use BSSID to set your target.

MRT-77 commented on WiFi/ESP8266/ESP8266.c:
```diff
@@ -1836,8 +1908,8 @@ int32_t AT_Resp_CurrentMode (uint32_t *mode) {
-int32_t AT_Cmd_ConnectAP (uint32_t at_cmode, const char *ssid, const char *pwd, const char *bssid) {
-  char out[64];
+int32_t AT_Cmd_ConnectAP (uint32_t at_cmode, const char *ssid, const char *pwd, const uint8_t *bssid) {
+  char out[113+1];
```
32 char SSID, 64 char Password, 17 char BSSID

MRT-77 commented on WiFi/ESP8266/ESP8266.c:
```diff
@@ -1848,7 +1920,7 @@ int32_t AT_Cmd_ConnectAP (uint32_t at_cmode, const char *ssid, const char *pwd,
 
     if (bssid != NULL) {
-      n += sprintf (&out[n], ",\"%s\"", bssid);
+      n += sprintf (&out[n], ",\"%02x:%02x:%02x:%02x:%02x:%02x\"", bssid[0], bssid[1], bssid[2], bssid[3], bssid[4], bssid[5]);
```
print bssid in "xx:xx:xx:xx:xx:xx" format

VladimirUmek commented:
This pull request has been partially merged into main repo, see commits
- 7f82dda76bb8c5a8a75e88db11ad6fcbf013f880 
- 4d20914c6c10f5eeec4248ddab7fa12ae8b6f11b
- 679caca3f9c19feeb5053b75f715ee4da0554152
- 60e0277c39876f115f203507111e5ae79d7b6bcd
- f874fa171b1a8af6ceafb7b01f298a72157b325e

Activate function used to connect to AP was fixed and now enables application to connect to AP with BSSID specified in SetOption, see 853a5139be12de48f759188ce64cda9908d1348d

So basically, everything that does not depend on CMSIS WiFi API has been merged. Other changes are still pending until they get discussed and approved/rejected/reworked.

MRT-77 committed:
Merge branch 'develop' into develop

MRT-77 committed:
Merge branch 'ARM-software:develop' into develop

MRT-77 committed:
Merge branch 'ARM-software:main' into develop

VladimirUmek commented:
Closing this issue since all was done from the CMSIS-Driver pack perspective. Further requests for API changes shall be opened in CMSIS_6.
